pkg/cmd/connector/create/create.go

```diff
@@ -46,62 +60,67 @@ func NewCreateCommand(f *factory.Factory) *cobra.Command {
 		},
 	}
 	flags := flagutil.NewFlagSet(cmd, f.Localizer)
-	flags.StringVar(&opts.file, "file", "", f.Localizer.MustLocalize("connector.file.flag.description"))
+	flags.StringVarP(&opts.file, "file", "", "f", f.Localizer.MustLocalize("connector.file.flag.description"))
+	flags.StringVar(&opts.kafkaId, "kafka", "", f.Localizer.MustLocalize("connector.flag.kafka.description"))
+	flags.StringVar(&opts.namespace, "namespace", "", f.Localizer.MustLocalize("connector.flag.namespace.description"))
+	flags.StringVar(&opts.name, "name", "", f.Localizer.MustLocalize("connector.flag.name.description"))
+	flags.BoolVar(&opts.serviceAccount, "create-service-account", false, f.Localizer.MustLocalize("connector.flag.sa.description"))
 	flags.AddOutput(&opts.outputFormat)
 
 	return cmd
 }
 
 func runCreate(opts *options) error {
 	f := opts.f
-
-	var conn connection.Connection
-	conn, err := f.Connection()
+	// Load the connector from the file
+	fileContent, err := readFileFromInput(opts)
 	if err != nil {
 		return err
 	}
 
-	var specifiedFile *os.File
-	if opts.file == "" {
-		opts.f.Logger.Info(opts.f.Localizer.MustLocalize("common.message.reading.file"))
-		file, newErr := util.CreateFileFromStdin()
-		if newErr != nil {
-			return newErr
-		}
-		specifiedFile = file
-	} else {
-		if util.IsURL(opts.file) {
-			specifiedFile, err = util.GetContentFromFileURL(f.Context, opts.file)
-		} else {
-			specifiedFile, err = os.Open(opts.file)
-		}
-		if err != nil {
-			return err
-		}
+	var userConnector connectormgmtclient.ConnectorRequest
+	err = json.Unmarshal(fileContent, &userConnector)
+	if err != nil {
+		return errors.Wrap(err, opts.f.Localizer.MustLocalize("connector.message.reading.file.error"))
 	}
-	defer specifiedFile.Close()
 
-	byteValue, err := ioutil.ReadAll(specifiedFile)
+	opts.f.Logger.Info(opts.f.Localizer.MustLocalize("connector.create.start"))
+
+	err = setDefaultValuesFromFlags(&userConnector, opts)
+	if err != nil {
+		return err
+	}
+
+	var conn connection.Connection
+	conn, err = f.Connection()
 	if err != nil {
 		return err
 	}
-	var connector connectormgmtclient.ConnectorRequest
-	err = json.Unmarshal(byteValue, &connector)
+
+	err = setValuesInInteractiveMode(&userConnector, opts, &conn)
 	if err != nil {
 		return err
 	}
 
 	api := conn.API()
 
 	a := api.ConnectorsMgmt().ConnectorsApi.CreateConnector(f.Context)
-	a = a.ConnectorRequest(connector)
+	a = a.ConnectorRequest(userConnector)
 	a = a.Async(true)
 
 	response, httpRes, err := a.Execute()
 	if httpRes != nil {
 		defer httpRes.Body.Close()
 	}
 
+	if apiErr := connectorerror.GetAPIError(err); apiErr != nil {
+		if apiErr.GetCode() == connectorerror.ERROR_7 {
```

Comment:
This is a possible issue with the backend, setting connector type to `data_generator_0.1_test` gets error with coce `CONNECTOR-MGMT-21`.